idf_build_apps/app.py

```diff
@@ -292,6 +296,35 @@ def size_json_path(self) -> t.Optional[str]:
     def config(self) -> t.Optional[str]:
         return self.config_name
 
+    def _process_override_sdkconfig_file(self):
+        self.override_sdkconfig_items = BuildAppJob.override_sdkconfig_items.copy()
+        override_sdkconfig_file_items = {}
+
+        for f in BuildAppJob.override_sdkconfig_files_name:
+            if not os.path.isabs(f):
+                f = os.path.join(self.work_dir, f)
+            if not os.path.isfile(f):
+                continue
+            with open(f) as fr:
+                for line in fr:
+                    m = self.SDKCONFIG_LINE_REGEX.match(line)
+                    if not m:
+                        continue
+                    override_sdkconfig_file_items[m.group(1)] = m.group(2)
+
+        if BuildAppJob.override_priority == OverridePriority.OVERRIDE_SDKCONFIG:
+            self.override_sdkconfig_items = override_sdkconfig_file_items | self.override_sdkconfig_items
+        else:
+            self.override_sdkconfig_items = self.override_sdkconfig_items | override_sdkconfig_file_items
```

Comment:
seems like the `override_priority` is quite confusing here

I recommend overriding `BuildAppJob.override_sdkconfig_items` with `App().override_sdkconfig_items`

(Please refer to the detailed overriding rules listed later)

Reply:
Updated according to recommendation.